lib/topics/index.js

```diff
@@ -57,6 +62,77 @@ Topics.prototype.getTopic = function(topicName, prefix, topicUid) {
 };
 
 /**
+ * Internal function to get Promises for done and error topics.
+ *
+ * @param doneTopicPromise  - A promise for the done topic.
+ * @param errorTopicPromise - A promise for the error topic.
+ * @returns {*}
+ */
+Topics.prototype.getTopicPromises = function getTopicPromises(doneTopicPromise, errorTopicPromise) {
+  var deferred = q.defer();
+
+  doneTopicPromise.then(function(createdItem) {
+    deferred.resolve(createdItem);
+  });
+
+  errorTopicPromise.then(function(error) {
+    deferred.reject(error);
+  });
+
+  return deferred.promise;
+};
+
+/**
+ *
+ * Internal function to get Promises for done and error topics.
+ *
+ * @param {Object} config - specific config settings such as timeout (if any)
+ * @param {string} topicName   - The name of the topic to generate
+ * @returns {Promise} - A promise for the topic.
+ */
+Topics.prototype.getErrorAndDoneTopicPromises = function getErrorAndDoneTopicPromises(config, topicName) {
+  //Creating a unique channel to get the response
+  // TODO move done/error prefixes here.
+
+  var topicUid = shortid.generate();
+
+  var doneTopic = this.getTopic(topicName, CONSTANTS.DONE_PREFIX, topicUid);
+  var errorTopic = this.getTopic(topicName, CONSTANTS.ERROR_PREFIX, topicUid);
+
+  var doneTopicPromise = this.mediator.promise(doneTopic);
+  var errorTopicPromise = this.mediator.promise(errorTopic);
+
+  var timeoutDefer = q.defer();
+
+  setTimeout(function() {
+    timeoutDefer.reject(new Error("Timeout For Topic: " + doneTopic));
+  }, config.topicTimeout || CONSTANTS.TOPIC_TIMEOUT);
+
+  //Either one of these promises resolves/rejects or it will time out.
+  return q.race([this.getTopicPromises(doneTopicPromise, errorTopicPromise), timeoutDefer.promise]);
+};
+
+
+Topics.prototype.execute = function getErrorAndDoneTopicPromises(topicName, payload, config) {
```

Comment:
Let's add this to the top or expose this method as this would be the most used function.